From 8f7fd227ff3a98cd051e9850b7ec6096ca0ca6cf Mon Sep 17 00:00:00 2001
From: troydm <d.geurkov@gmail.com>
Date: Fri, 25 May 2012 13:15:36 +0400
Subject: [PATCH] modified r1

---
 wininput.c |   13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/wininput.c b/wininput.c
index d934ac7..464c890 100644
--- a/wininput.c
+++ b/wininput.c
@@ -455,6 +455,8 @@ win_key_down(WPARAM wp, LPARAM lp)
   void ss3(char c) { ch('\e'); ch('O'); ch(c); }
   void csi(char c) { ch('\e'); ch('['); ch(c); }
   void mod_csi(char c) { len = sprintf(buf, "\e[1;%c%c", mods + '1', c); }
+  void mod_csi2(int m, char c) { len = sprintf(buf, "\e[1;%d%c", m, c); }
+  void mod_tilde3(int a, int b, int c) { len = sprintf(buf, "\e[%d;%d;%d~", a,b,c); }
   void mod_ss3(char c) { mods ? mod_csi(c) : ss3(c); }
   void tilde_code(uchar code) {
     len = sprintf(buf, mods ? "\e[%i;%c~" : "\e[%i~", code, mods + '1');
@@ -505,8 +507,12 @@ win_key_down(WPARAM wp, LPARAM lp)
   }
   
   void cursor_key(char code, char symbol) {
-    if (!app_pad_key(symbol))
-      mods ? mod_csi(code) : term.app_cursor_keys ? ss3(code) : csi(code);
+    if (!app_pad_key(symbol)){
+        if(shift && alt)
+            mod_csi2(10,code);
+        else
+            mods ? mod_csi(code) : term.app_cursor_keys ? ss3(code) : csi(code);
+    }
   }
 
   // Keyboard layout
@@ -657,6 +663,8 @@ win_key_down(WPARAM wp, LPARAM lp)
         return 0;
       if (!ctrl)
         shift ? csi('Z') : ch('\t');
+      else if (ctrl) 
+          shift ? mod_tilde3(27,6,9) : mod_tilde3(27,5,9);
       else if (cfg.switch_shortcuts) {
         win_switch(shift);
         return 1;
@@ -683,6 +691,7 @@ win_key_down(WPARAM wp, LPARAM lp)
             28, 29, 31, 32, 33, 34, 42, 43, 44, 45
           }[key - VK_F5]
         );
+
       }
     when VK_INSERT: edit_key(2, '0');
     when VK_DELETE: edit_key(3, '.');
-- 
1.7.10.msysgit.1

